### PR TITLE
Item 9448:  Remove media sample types from samples for freezers

### PR DESCRIPTION
**Pull request by XingY**
#### Rationale
Biologics has 2 special samples types that are "media". So far, name matching has been used to determine if a sample type is media or not. This PR adds a Category field to exp.MaterialSource, to allow "media" category samples.

#### Related Pull Requests
* https://github.com/LabKey/platform/pull/2664
* https://github.com/LabKey/inventory/pull/315
* https://github.com/LabKey/sampleManagement/pull/718
* https://github.com/LabKey/biologics/pull/1018
* https://github.com/LabKey/labkey-ui-components/pull/648

#### Changes
* Add Category field to exp.MaterialSource

### CODE REVIEW

File: api/src/org/labkey/api/exp/query/ExpSchema.java [hunk 1]
```diff
@@ -340,6 +342,11 @@ public TableInfo createTable(String name, ContainerFilter cf)
             return new EnumTableInfo<>(SampleStateType.class, this, SampleStateType::name, true, "Contains the available sample state (status) types.");
         }
 
+        if (SAMPLE_TYPE_CATEGORY_TABLE.equalsIgnoreCase(name))
+        {
+            return new EnumTableInfo<>(SampleTypeCategoryType.class, this, SampleTypeCategoryType::name, true, "Contains the list of available sampletype category types.");
```
Comment (cnathe) [inline api/src/org/labkey/api/exp/query/ExpSchema.java]:
```suggestion
            return new EnumTableInfo<>(SampleTypeCategoryType.class, this, SampleTypeCategoryType::name, true, "Contains the list of available sample type category types.");
```

File: experiment/resources/schemas/exp.xml [hunk 2]
```diff
@@ -388,8 +395,8 @@
       <column columnName="Category">
         <fk>
             <fkDbSchema>exp</fkDbSchema>
-            <fkTable>DataClassCategory</fkTable>
-            <fkColumnName>category</fkColumnName>
+            <fkTable>DataClassCategoryType</fkTable>
+            <fkColumnName>Value</fkColumnName>
```
Comment (cnathe) [inline experiment/resources/schemas/exp.xml:399]: huh, so was this lookup never resolving in the UI until now?

File: experiment/src/org/labkey/experiment/XarExporter.java [hunk 3]
```diff
@@ -646,6 +646,11 @@ else if (sampleType.hasIdColumns())
             xSampleSet.setMetricUnit(sampleType.getMetricUnit());
         }
 
+        if (sampleType.getCategory() != null)
+        {
+            xSampleSet.setCategory(sampleType.getCategory());
```
Comment (cnathe) [inline experiment/src/org/labkey/experiment/XarExporter.java:651]: maybe I missed it, but is the folder export/import case in the test plan?

**Review by cnathe — APPROVED** [review]

File: experiment/src/org/labkey/experiment/XarExporter.java (hunk 3 above)
Comment (XingY) [reply experiment/src/org/labkey/experiment/XarExporter.java:651]: good point, added in test plan

**Pull request merged**